migration to PhotoSwipe v5

kfsz opened the pull request:
Hey, so I tried to get this library working with PhotoSwipe v5 and this is what I ended up with. 

Everything seems to work (and work great), but:
* there are no type definitions for PhotoSwipe 5 right now - so, sorry for the `any`s ^^
*  I haven't made any updates to docs - I could try updating them if you'd like me to though
  
Also PS v4 -> PS v5 had some big breaking changes, so here's a short migration guide:
1. remove `photoswipeRef` from `natural-gallery-js` init, it's no longer needed anywhere
2. neither is photoswipe/lightbox html template, so remove that also 
  (https://ecodev.github.io/natural-gallery-js/docs-options.html#lightbox)
4. check which options are available in PS v5 (https://photoswipe.com/options/)
  
after that PhotoSwipe should load  ^^   

Here are changes I made to API:
1. added PS getter, so you can now access PS with `naturalGalleryInstance.photoSwipe`
  (which can be then used to listen to events, add UI buttons and other things)
2. also added current PS item getter (`naturalGalleryInstance.photoSwipeCurrentItem`)
3. removed `zoom` event dispatch as it seemed redundant: 
    * PS already provides open events (https://photoswipe.com/events/)
    * and PS instance is now always accessible via  `naturalGalleryInstance.photoSwipe`
  
Gonna add few comments to `files changed` to explain why I made some internal changes.  
And I think that should be all - please take a look and check if it'd be okay to merge.  

resolves #75

kfsz commented on src/js/Item.ts:
```diff
@@ -102,7 +102,7 @@ export class Item<Model extends ModelAttributes> {
 
         const element = this.document.createElement('div') as HTMLElement;
-        let image: HTMLElement = this.document.createElement('div');
+        let image: HTMLElement = this.document.createElement('a');
```
photoswipe recommended gallery markup  
(https://photoswipe.com/getting-started/#required-html-markup)  
` Each element that matches the selector should be or should contain link <a> element`  

actually, I think it may no longer be required as I later changed to itemData filter as PS images source (https://photoswipe.com/data-sources/#dynamically-generated-data) so it's possibly safe to remove - I think it'd cleaner to keep it as PS suggests though

kfsz commented on src/js/Item.ts:
```diff
@@ -250,7 +250,7 @@ export class Item<Model extends ModelAttributes> {
         img.setAttribute('src', this.model.thumbnailSrc);
 
-        this._image.style.backgroundImage = 'url(' + this.model.thumbnailSrc + ')';
+        this._image.appendChild(img);
```
again, PS recommended html markup (https://photoswipe.com/data-sources/#custom-html-markup)  
`<img>` here is required for zoom transition (nice thumbnail pic to full page pic animation)  

as docs say it could be also done via `domItemData` but simply using <img> seems way cleaner

kfsz commented on src/styles/_figure.scss:
```diff
@@ -42,6 +42,12 @@
+        .image.zoomable:active {
+            transform:scale(1.0);
+            transition: 1s ease-out opacity, 0.1s ease-out transform, .2s ease-out margin;
+        }
```
makes zoom transition (from previous comment) look more natural and gives some feedback to clicking the image  
without it, zoom transition seems kinda weird, it changes instantly to initial size, and then zooms in - and, once thumbnail loses focus it slowly shrinks back to normal size again, while full screen image is already open

sambaptista commented:
I've deployed your fork locally, but doesn't work when trying to open photoswipe.

![capture 2022-05-24 at 20 21 46](https://user-images.githubusercontent.com/5111137/170105891-a63eef9a-833a-49de-be27-7a9f05a7eb0a.png)

Does it work when you deploy it yourself + `yarn dev` + `127.0.0.1:4000` ?

kfsz commented:
@sambaptista 

yeah, I completely skipped homepage last time. fixed it today (and lots of other issues I noticed along the way :stuck_out_tongue: ) - should work as expected now.  

with exception of captions - they are no longer supported out of the box (https://photoswipe.com/caption/), but it's possible to implement them via API or plugins - I added a quick and dirty example how to do that for Natural Gallery (`index.html`), skipping others, as I can't decide if it's better with or without them - I think I'd skip them on main gallery pages and maybe add an example on how to use them in `Theming and customisation`, not sure though.

also, still saving documentation part for later, but besides that, everything should work, I think.

sambaptista commented:
OK; It's finally done, I've reviewed the code and tested it. Thanks a lot for your contribution. 

Almost everything is ok. I will do a few adjustments on my side and merge your two pull requests.

Would you have time to fix the tests ? E2E are ok, but UNIT fail. The issue is about the import below. 

```
yarn test unit
```

```
    Cannot find module 'photoswipe/lightbox' from 'src/js/galleries/AbstractGallery.ts'
```

Keep me informed about your intentions to fix it (or not).

PowerKiKi commented:
@kfsz, if you push new commits to fix tests, then the CI _should_ trigger and validate your changes. If it doesn't trigger, please let us know and we'll investigate why it didn't trigger.

kfsz commented:
@sambaptista @PowerKiKi 

tests fixed, they should work (and pass) now.

workflow needs approval though, so please approve.

PowerKiKi commented:
Thanks, I approved the workflow, and it did pass 🎉 

I'll let @sambaptista do the merging.
